File: src/test/java/no/nav/familie/ks/oppslag/felles/config/StsTestConfig.java
package no.nav.familie.ks.oppslag.felles.config;

import no.nav.familie.ks.oppslag.felles.rest.StsRestClient;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@Configuration
public class StsTestConfig {
    @Bean
    @Profile("mock-sts")
    @Primary
    public StsRestClient stsRestClientMock() {
        StsRestClient client = mock(StsRestClient.class);

        when(client.getSystemOIDCToken()).thenReturn("MOCKED-OIDC-TOKEN");
        return client;
    }
}

File: src/test/java/no/nav/familie/integrasjoner/felles/config/StsTestConfig.kt
package no.nav.familie.integrasjoner.felles.config

import io.mockk.every
import io.mockk.mockk
import no.nav.familie.felles.tokenklient.sts.StsRestClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile

@Configuration
class StsTestConfig {
    @Bean
    @Primary
    @Profile("mock-sts")
    fun stsRestClientMock(): StsRestClient {
        val client = mockk<StsRestClient>()
        every { client.systemOIDCToken } returns "MOCKED-STS-OIDC-TOKEN"
        return client
    }
}
